x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -142,7 +143,7 @@ protected LogicalPlan rule(EsqlUnresolvedRelation plan, AnalyzerContext context)
             EsIndex esIndex = context.indexResolution().get();
             var attributes = mappingAsAttributes(plan.source(), esIndex.mapping());
             attributes.addAll(plan.metadataFields());
-            return new EsRelation(plan.source(), esIndex, attributes);
+            return attributes.isEmpty() ? new EmptyEsRelation(esIndex) : new EsRelation(plan.source(), esIndex, attributes);
```

Comment:
To Luigi's serialisation concern, I guess this could be smth like
```suggestion
            return new EsRelation(plan.source(), esIndex, attributes.isEmpty() ? NO_FIELDS : attributes);
```
with `s/EMPTY_OUTPUT/NO_FIELDS`?